src/components/PatientForm.tsx

```diff
@@ -25,10 +25,24 @@ const PatientForm: FunctionComponent<PatientFormProps> = ({ patient }) => {
   const navigate = useNavigate()
 
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
-    const fieldName: string = e.target.name
     const fieldValue: string = e.target.value
-    const newField = { [fieldName]: fieldValue }
-    setPatientState({ ...patientState, ...newField })
+    const keys = e.target.name.split('.')
+    const fieldName = keys[0]
+    if (fieldName in patientState) {
+      const field = patientState[fieldName as keyof typeof patientState]
+      if (field instanceof Object) {
+        const nextValue = [...field]
+        // @ts-expect-error
+        nextValue[keys[1]][keys[2]] = fieldValue
```

Comment:
Looks very fishy.
Maybe for the prescription you want to create a dedicated state if that makes it easier to manage

Reply:
yes I wanted a dedicated state but in doubt of your answer I made a single state

Reply:
Fixed in https://github.com/issa-diallo/mynotif_frontend/pull/114